### PR TITLE
When trying to exchange the SameProduct, in dropdown display also backorderable items, not only in_stock.

### LINKED ISSUE #6575: Unable to exchange for out of stock items that are backorderable
When creating a new return authorization, the "exchange for" dropdown only shows variants of the product that's in stock, even if the out of stock items are marked as backorderable. 

To reproduce in the sandbox store: 
1. pick a product that has multiple variants
2. make one of the variants out of stock and check both `track inventory` and `backorderable`
3. place an order of this product (any variant)
4. ship it from order admin page
5. go to returns authorization page and create a New Return Authorization 
-> the out of stock variant doesn't show up in the "exchange for" dropdown

It my understanding that if an item is backorderable, no matter what the inventory looks like, one should be able to order it and/or exchange for it, right?

### PR DESCRIPTION
**Pull request by dimon-k**
Fix #6575

### CODE REVIEW

File: core/app/models/spree/variant.rb [hunk 1]
```diff
@@ -56,6 +56,12 @@ class Variant < Spree::Base
     after_touch :clear_in_stock_cache
 
     scope :in_stock, -> { joins(:stock_items).where('count_on_hand > ? OR track_inventory = ?', 0, false) }
+    scope :in_stock_or_backorderable, -> do
+      joins(:stock_items).where('count_on_hand > ? OR
+                                 track_inventory = ? OR
```
Comment (bbonislawski) [inline core/app/models/spree/variant.rb]: Don't use `?` inside query. Params are not dynamic and you can just pass them there

**Review by bbonislawski — CHANGES_REQUESTED** [review]

File: core/app/models/spree/variant.rb (hunk 1 above)
Comment (damianlegawiec) [reply core/app/models/spree/variant.rb]: @dimon-k you can just use ActiveRecord's `or` method

https://blog.bigbinary.com/2016/05/30/rails-5-adds-or-support-in-active-record.html

Comment (dimon-k) [reply core/app/models/spree/variant.rb]: Fixed.

**Review by kushniryb — CHANGES_REQUESTED** [review]

File: core/app/models/spree/variant.rb [hunk 2]
```diff
@@ -56,6 +56,8 @@ class Variant < Spree::Base
     after_touch :clear_in_stock_cache
 
     scope :in_stock, -> { joins(:stock_items).where('count_on_hand > ? OR track_inventory = ?', 0, false) }
+    scope :backorderable, -> { joins(:stock_items).where('spree_stock_items.backorderable' => true) }
```
[the remark this answers is not in the document]
Comment (dimon-k) [reply core/app/models/spree/variant.rb]: Sure we can. Changed!

**Review by kushniryb — APPROVED** [review]

File: core/spec/models/spree/return_item/exchange_variant_eligibility/same_product_spec.rb [hunk 3]
```diff
@@ -28,12 +28,16 @@ module ReturnItem::ExchangeVariantEligibility
           expect(SameProduct.eligible_variants(variant)).not_to include other_product_variant
         end
 
-        it 'only returns variants that are on hand' do
-          product = create(:product, variants: Array.new(2) { create(:variant) })
+        it 'only returns variants that are on hand or backorderable' do
+          product = create(:product, variants: Array.new(3) { create(:variant) })
```
Comment (VygovskySergey) [inline core/spec/models/spree/return_item/exchange_variant_eligibility/same_product_spec.rb]: I believe that here you can use `create_list(:variant, 3)`

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**